spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/GatewayRedisAutoConfiguration.java

```diff
@@ -43,6 +44,8 @@
 @AutoConfigureBefore(GatewayAutoConfiguration.class)
 @ConditionalOnBean(ReactiveRedisTemplate.class)
 @ConditionalOnClass({ RedisTemplate.class, DispatcherHandler.class })
+@ConditionalOnProperty(name = "spring.cloud.gateway.redis-auto-configuration.enabled",
```

Comment:
Change to `spring.cloud.gateway.redis.enabled`

Reply:
Ok